imod/mf6/wel.py

```diff
@@ -410,242 +688,280 @@ def __create_assigned_wells(
-            2D DataArray with a ``ncellid`` rows and 3 to 2 columns, depending
-            on whether on a structured or unstructured grid."""
+        return cls(
+            x=np.array(x, dtype=float),
+            y=np.array(y, dtype=float),
+            screen_top=np.array(filt_top, dtype=float),
+            screen_bottom=np.array(filt_bot, dtype=float),
+            rate=well_rate,
+            minimum_k=minimum_k,
+            minimum_thickness=minimum_thickness,
+        )
 
-        # Find indices belonging to x, y coordinates
-        indices_cell2d = points_indices(dst_grid, out_of_bounds="ignore", x=x, y=y)
-        # Convert cell2d indices from 0-based to 1-based.
-        indices_cell2d = {dim: index + 1 for dim, index in indices_cell2d.items()}
-        # Prepare layer indices, for later concatenation
 
-        if isinstance(dst_grid, xu.UgridDataArray):
-            indices_layer = xr.DataArray(
-                layer, coords=indices_cell2d["mesh2d_nFaces"].coords
-            )
-            face_dim = dst_grid.ugrid.grid.face_dimension
-            indices_cell2d_dims = [face_dim]
-            cell2d_coords = ["cell2d"]
-        else:
-            indices_layer = xr.DataArray(layer, coords=indices_cell2d["x"].coords)
-            indices_cell2d_dims = ["y", "x"]
-            cell2d_coords = ["row", "column"]
+class LayeredWell(GridAgnosticWell):
+    """
+    Agnostic WEL package, which accepts x, y and layers.
 
-        # Prepare cellid array of the right shape.
-        cellid_ls = [indices_layer] + [
-            indices_cell2d[dim] for dim in indices_cell2d_dims
-        ]
-        cellid = xr.concat(cellid_ls, dim="nmax_cellid")
-        # Rename generic dimension name "index" to ncellid.
-        cellid = cellid.rename(index="ncellid")
-        # Put dimensions in right order after concatenation.
-        cellid = cellid.transpose("ncellid", "nmax_cellid")
-        # Assign extra coordinate names.
-        coords = {
-            "nmax_cellid": ["layer"] + cell2d_coords,
-            "x": ("ncellid", x),
-            "y": ("ncellid", y),
-        }
-        cellid = cellid.assign_coords(**coords)
+    This package can be written to any provided model grid.
+    Any number of WEL Packages can be specified for a single groundwater flow model.
+    https://water.usgs.gov/water-resources/software/MODFLOW-6/mf6io_6.0.4.pdf#page=63
 
-        return cellid
+    Parameters
+    ----------
 
-    def render(self, directory, pkgname, globaltimes, binary):
-        raise NotImplementedError(
-            f"{self.__class__.__name__} is a grid-agnostic package and does not have a render method. To render the package, first convert to a Modflow6 package by calling pkg.to_mf6_pkg()"
-        )
+    y: float or list of floats or np.array of floats
+        is the y location of the well.
+    x: float or list of floats or np.array of floats
+        is the x location of the well.
+    layer: int or list of ints or np.array of ints
```

Comment:
is there one layer per well or can there be multiple? If we assign 1 layer as an int,  will it be used for all wells?

Reply:
Scalars are not supported anymore. Docstring had to be updated